src/components/cesium_map/api/query.js

```diff
@@ -0,0 +1,150 @@
+    `(${field.lowerBound}:[* TO ${filters[0]}] AND ${field.upperBound}:[${filters[1]} TO *])`
+  );
+};
+
+const listFacetFieldToQueryFilter = (field) => {
+  const filters = field.value || [];
+  if (filters.length === 0) {
+    return null;
+  }
+
+  const filterQ = filters.map((f) => `"${f}"`).join(" OR ");
+  return encodeURIComponent(`${field.field}:(${filterQ})`);
+};
+
+const textFieldToQueryFilter = (field) => {
+  if (!field.value || field.value.length === 0) {
+    return null;
+  }
+
+  return encodeURIComponent(`${field.field}:${field.value}`);
+};
+
+const fieldToQueryFilter = (field) => {
+  if ((field.type === "text" || field.type === "non-search") && field.field !== "*") {
+    return textFieldToQueryFilter(field);
+  } else if (field.type === "list-facet") {
+    return listFacetFieldToQueryFilter(field);
+  } else if (field.type === "range-facet" || field.type === "range" || field.type === "date-range-facet") {
+    return rangeFacetToQueryFilter(field);
+  } else if (field.type === "period-range-facet" || field.type === "period-range") {
+    return periodRangeFacetToQueryFilter(field);
+  }
+  return null;
+};
+
+const buildFormat = (format) => Object.keys(format)
+  .map((key) => `${key}=${encodeURIComponent(format[key])}`)
+  .join("&");
+
+const buildQuery = (fields) => fields
+  .map(fieldToQueryFilter)
+  .filter((queryFilter) => queryFilter !== null)
+  .map((queryFilter) => `fq=${queryFilter}`)
+  .join("&");
+
+// Concates all request fields for solr fl parameters
+const requestField = (fields) => fields
+  .map((field) => `${encodeURIComponent(field.field)}`)
+  .join(" ")
+
+const dateRangeFacetFieldValue = (dateRangeFacetField) => {
+  var filters = dateRangeFacetField.value;
+  if (!filters || filters.length === 0) {
+    return "";
+  }
+  const dateSuffix = "-01-01T00:00:00Z";
+  // startRange will be year-only, format in the way that makes solr happy
+  let startRangeValue = filters[0] + dateSuffix;
+  let endRangeValue = filters[1] + dateSuffix;
+  return `facet.range=${encodeURIComponent(dateRangeFacetField.field)}&facet.range.gap=${encodeURIComponent("+1YEARS")}&facet.range.start=${startRangeValue}&facet.range.end=${endRangeValue}`;
+};
+
+const facetFields = (fields) => fields
+  .filter((field) => field.type === "list-facet" || field.type === "range-facet")
+  .map((field) => `facet.field=${encodeURIComponent(field.field)}`)
+  .concat(
+    fields
+      .filter((field) => field.type === "period-range-facet")
+      .map((field) => `facet.field=${encodeURIComponent(field.lowerBound)}&facet.field=${encodeURIComponent(field.upperBound)}`)
+  )
+  .concat(
+    fields
+      .filter((field) => field.type === "date-range-facet")
+      .map((field) =>
+        dateRangeFacetFieldValue(field))
+  )
+  .join("&");
+
+const facetSorts = (fields) => fields
+  .filter((field) => field.facetSort)
+  .map((field) => `f.${encodeURIComponent(field.field)}.facet.sort=${field.facetSort}`)
+  .join("&");
```

Comment:
These first 100 lines look similar to code that's in the solr client portion of `solr-faceted-search-react`.  Are they identical?  If they are, could we look at calling those functions rather than copy/pasting here?

Reply:
The first 100 lines are the same but the solr-faceted-search-react library doesn't export those functions. I will modify the library to expert it. Do you think it's ok?

Reply:
I did a similar thing to export the date range sliders.  I think exporting the functions is better than doing a wholesale copy and paste.  Do you agree, @datadavev?

Reply:
Sure. An export is a bit simpler from a maintenance perspective